src/Modal/Modal.jsx

```diff
@@ -62,11 +101,41 @@ Modal.propTypes = {
     PropTypes.node
   ]).isRequired,
   onClose: PropTypes.func,
+  onCancel: PropTypes.func,
+  onSubmit: PropTypes.func,
+  cancelText: ({ onCancel, cancelText }) => {
+    if (onCancel && !cancelText) {
+      return new Error(
+        "You might need to add property 'cancelText' if you need the button to have text"
+      );
+    }
+    if (typeof cancelText !== "string") {
+      return new Error(`cancelText needs to be a string`);
+    }
+
+    return null;
+  },
+  submitText: ({ onSubmit, submitText }) => {
+    if (onSubmit && !submitText) {
+      return new Error(
+        "You might need to add property 'submitText' if you need the button to have text"
+      );
+    }
+    if (typeof submitText !== "string") {
+      return new Error(`submitText needs to be a string`);
+    }
+
+    return null;
+  },
   role: PropTypes.string,
   ariaLabel: PropTypes.string
 };
 Modal.defaultProps = {
-  onClose: () => {},
+  onClose: null,
+  onCancel: null,
+  onSubmit: null,
+  cancelText: "",
+  submitText: "",
```

Comment:
These text props shouldn't have default blank string values anymore right? As they are basically 'required' when onsubmit is passed

Reply:
what do you think it should be?

Reply:
no default? we are throwing a prop type warning when its not set

Reply:
eslint complains since it is not always required but only sometimes.. any ideas?

Reply:
Ohhh it doesnt work well with custom prop validators I guess then... can we change them to null so it's a bit more obvious they are not usable as strings?

Reply:
When making it `null`  and because we have a custom validation that would check if the user adds as property a `string` an error will be thrown that the specific prop needs to be string and not null.

Why dont we keep it empty as it is since its an empty string ?

Reply:
ok lol, no big deal. 

Im beginning to dislike the 'optional props' props should have default rule

Reply:
yes i remember you told me that. websync